### PR TITLE
[SYCL] Add the experimental query interface for joint matrix

### PR DESCRIPTION
**Pull request by dkhaldi**
Signed-off-by: Dounia <dounia.khaldi@intel.com>

### CODE REVIEW

File: sycl/include/sycl/ext/oneapi/matrix/static-query.hpp [hunk 1]
```diff
@@ -0,0 +1,474 @@
...
+    matrix_type btype;
+    matrix_type ctype;
+    uint32_t msize;
+    uint32_t nsize;
+    uint32_t ksize;
+  };
+  static constexpr int num_combinations = 2;
+  combination combinations[num_combinations];
+  constexpr tpu_params() : combinations() {
+    //Note that unsigned int8 variants are not implemented yet
```
Comment (LuoYuanke) [inline sycl/include/sycl/ext/oneapi/matrix/static-query.hpp:99]: // Note

File: sycl/include/sycl/ext/oneapi/matrix/static-query.hpp [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+
+// Specialization for when only types are given, need to query only sizes
+template <typename Ta, typename Tb, typename Tc>
+struct tpu_params<tpu::amx, Ta, Tb, Tc, 0, 0, 0,
+                  typename std::enable_if<(!std::is_same_v<Ta, void> &&
+                                           !std::is_same_v<Tb, void> &&
+                                           !std::is_same_v<Tc, void>)>::type> {
+  static_assert((are_types_valid_amx<Ta, Tb, Tc>()),
+                "Invalid types for AMX, supported types are int8_t, "
+                "half, and bf16");
```
[the remark this answers is not in the document]
Comment (dkhaldi) [reply sycl/include/sycl/ext/oneapi/matrix/static-query.hpp:123]: half is not, I will add a note that bf16 is supported using unsigned short data type.

File: sycl/include/sycl/ext/oneapi/matrix/static-query.hpp [hunk 3]
```diff
@@ -0,0 +1,474 @@
...
+      combinations[i + ii].ksize = ((sizeof(Ta) == 1) ? 32 : 16);
+      combinations[i + ii].nsize = 8;
+    }
+  }
+};
+
+// Specialization when both types and sizes are given
+template <typename Ta, typename Tb, typename Tc, int M, int N, int K>
+struct tpu_params<
+    tpu::dpas, Ta, Tb, Tc, M, N, K,
```
[the remark this answers is not in the document]
Comment (dkhaldi) [reply sycl/include/sycl/ext/oneapi/matrix/static-query.hpp:389]: clang-format did that

**Review by LuoYuanke — DISMISSED** [review]
LGTM, I assume constexpr indicate compiler the value is constant.

**Review by LuoYuanke — APPROVED** [review]

**Comment by dkhaldi** [comment]
@bader Can you help review and merge?

**Review by dm-vodopyanov — APPROVED** [review]

**Pull request merged**